### PR TITLE
Add email troubleshooting guide

### PR DESCRIPTION
**Pull request by dotproto**
This PR introduces a new section to the [Developer account](https://extensionworkshop.com/documentation/publish/developer-accounts/) page that covers troubleshooting issues receiving emails from AMO.

### CODE REVIEW

File: src/content/documentation/publish/developer-accounts.md [hunk 1]
```diff
@@ -59,5 +59,32 @@ If you believe your account has been incorrectly blocked, please email amo-admin
...
+
+1. The email is sitting in a spam or junk folder in the recipient's inbox.
+    * To remedy this, you can add the AMO email domain (@mozilla.org) to your [approved senders](https://clean.email/blog/email-security/how-to-whitelist-an-email) list.
+1. The email was blocked from being delivered by a security filter. In this case, the email was filtered after being received by the email server – it never even reached the recipient's personal inbox.
+
+Another potential issue is if the primary email address associated with your Mozilla account no longer exists or if you no longer have access to it. In this case, please follow the instructions [here](https://support.mozilla.org/en-US/kb/change-primary-email-address-firefox-accounts) to update the email address on your Mozilla account.
+
+One final item to check is whether or not your inbox is full. If it is, you will need to delete some messages before you are able to start receiving emails again.
+
+If you are still experiencing problems receiving emails from the AMO platform we encourage you to post in our [add-ons community forum](https://discourse.mozilla.org/c/add-ons/35) with the specific details of your issue.
```
Comment (rebloor) [inline src/content/documentation/publish/developer-accounts.md:80]: Can we first clarify why someone would become aware of the fact that they haven't received emails? 

Then could we simplify e.g.:
```suggestion
If you're expecting but haven't received emails from the https://addons.mozilla.org (AMO) platform, check that:

1. Your Mozilla account's primary email address is correct. To update it, see the support article [Change primary email address on Mozilla account](https://support.mozilla.org/en-US/kb/change-primary-email-address-firefox-accounts).
1. The email isn't in your spam or junk folder. If it is, consider adding the AMO email domain (@mozilla.org) to your [approved senders](https://clean.email/blog/email-security/how-to-whitelist-an-email) list.
1. Your inbox isn't full.
1. Your mail provider hasn't implemented a security filter that blocked the email delivery.

If you're still experiencing issues after making these checks, posting details of your issue to the [add-ons community forum](https://discourse.mozilla.org/c/add-ons/35) may help.
```

**Comment by esullivan-moz** [comment]
Regarding clarifying why someone would become aware of the fact that they haven't been receiving emails, this is part of a larger workflow where the user has context.

File: src/data/pages.json [hunk 2]
```diff
@@ -885,6 +885,10 @@
                   {
                     "title": "Blocked accounts",
                     "id": "blocked-accounts"
+                  },
+                  {
+                    "title": "Issues receiving reviewer emails",
```
Comment (wagnerand) [inline src/data/pages.json:890]:
```suggestion
                    "title": "Issues receiving emails from AMO",
```
Or something around that. It's not just about reviewer emails.

**Comment by rebloor** [comment]
> Regarding clarifying why someone would become aware of the fact that they haven't been receiving emails, this is part of a larger workflow where the user has context.

@esullivan-moz the point I meant to make is "Do we clearly indicate within the process descriptions at the developer should now receive an email and link to this content with the note 'If you don't receive an email, check Issues receiving emails from AMO"

**Comment by wagnerand** [comment]
@dotproto is this the same copy as in the Google doc?

**Comment by dotproto** [comment]
@wagnerand, it was identical in [first commmit](https://github.com/mozilla/extension-workshop/pull/1850/commits/3e5387c2dbcdb6c68fe0d0f739ed139fa885315c) but it diverged after accepting changes in this PR.

**Review by wagnerand — APPROVED** [review]
I assume the actual enumeration numbers will be auto-generated, as every item uses `1.`.

Looks good to me overall, but @esullivan-moz should confirm since he drafted the copy for the page.

**Pull request merged**